### ISSUE TITLE
It's back to login screen when login my nextcloud server

### ISSUE THREAD

**Issue by akong77** [nextcloud/android#2609]
### Actual behaviour
It's back to login screen when login my nextcloud server

### Expected behaviour
It's back to login screen when login my nextcloud server
 
### Steps to reproduce
1. input my nextcloud server address.
2. input account and password
3. It's will back to need input server address screen

### Environment data
Android version: 8.0

Device model:  HTC 10

Stock or customized system: No

Nextcloud app version: 3.1.0

Nextcloud server version: 13.0.2

#### Web server error log
```
192.168.2.1 - - [23/May/2018:22:30:42 +0800] "GET /status.php HTTP/1.1" 200 146 "-" "Mozilla/5.0 (Android) ownCloud-android/3.1.0"
192.168.2.1 - - [23/May/2018:22:30:43 +0800] "HEAD /remote.php/webdav/ HTTP/1.1" 401 - "-" "Mozilla/5.0 (Android) ownCloud-android/3.1.0"
192.168.2.1 - - [23/May/2018:22:30:43 +0800] "GET /status.php HTTP/1.1" 200 146 "-" "Mozilla/5.0 (Android) ownCloud-android/3.1.0"
192.168.2.1 - - [23/May/2018:22:30:43 +0800] "HEAD /remote.php/webdav/ HTTP/1.1" 401 - "-" "Mozilla/5.0 (Android) ownCloud-android/3.1.0"
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Is this in an upgrade from a previous version?

Can you create us a test account, test if the problem occurs also there and if so send the credentials to tobias at nextcloud dot com with a reference to this issue?

**Comment by ewan86** [comment]
I have this issue too, I have a one plus two , running lineage OS 15.1 and nextcloud app version 3.1.0. 

I have a free account with owndrive and this has been happening to me for a while, I can fix it by uninstalling the app and then reinstalling.

**Comment by pbuyle** [comment]
I've the same issue, running Lineage OS 14.1 and app 3.1.0 (installed from F-Droid). I was able to successfully login after installing the app. But after rebooting the phone, the account was lost and the application restarted from the _server address_ screen. When credentials are provided, no error message is displayed. The expected "Redirecting" message is show for a brief moment before looping back to the _server address_ screen.

Re-installing the app does not help.

Weirdly, I've the exact same issue with the Seafile Android application. In https://forum.seafile.com/t/solved-android-app-account-disappears/4947/14, it is mentioned that this is an Android issue with how and when it clean up dead account which need access to the SD card. However, I tried formatting the SD card as a external storage (so the app is not installed on the SD card, and the SD card is not the primary storage for any synced folder). This didn't solve the issue.
